### ISSUE TITLE
When i create a Tree , its not maintaining the order which in i created the children when i invoke show method

### ISSUE THREAD

**Issue by salman0149** [caesar0301/treelib#170]
>>> from treelib import Node, Tree
>>> tree = Tree()
>>> tree.create_node('Students', 'Students', parent=None)
Node(tag=Students, identifier=Students, data=None)
>>> tree.create_node('Ben', 'Ben',parent='Students')
Node(tag=Ben, identifier=Ben, data=None)
>>> tree.create_node('Annie', 'Annie',parent='Students')
Node(tag=Annie, identifier=Annie, data=None)
>>> tree.show()
Students
├── Annie
└── Ben

As you see eventhough i created Ben Node first as child it is displaying Annie first, is there an argument in show() or in create_method() which i can pass to maintain the order when i call show method

**Comment by unikevin** [comment]
This is not an ordered tree library but an unordered tree lib.
I'm looking for a good tree lib like this (but ordered)😭

**Comment by frederik-elwert** [comment]
@unikevin: Shouldn’t this be `expand_tree(sorting=False)`? It’s a bit confusing, but from what I gathered, `sorting=True` means that sorting should happen based on some sort key, while `sorting=False` means it does not sort the nodes but maintains insertion order.

Also, I guess it would be nice if the same parameter was available on the `show()` method, because I also cannot find a way to display the tree while maintaining insertion order.

**Comment by frederik-elwert** [comment]
@salman0149 I think it should be possible to pass a custom counter as a data property and then sort by that key in `show()`.

**Comment by HollowMan6** [comment]
Hi! I've submitted a PR #180. If it's merged, you can use `show(sorting=False)` to display the tree while maintaining insertion order.

**Comment by salman0149** [comment]
Thanks very much , sorry for the late reply let me try this out

**Comment by salman0149** [comment]
On a seperate node when i pass tree.show(key=False) , it preserves the order of insertion.

**Comment by christian-krieg** [comment]
Hey guys! Without having a look into existing pull requests, I also implemented this functionality in https://github.com/caesar0301/treelib/pull/188 and https://github.com/caesar0301/treelib/pull/189. Unlike https://github.com/HollowMan6, I removed legacy ordering; Is there any reason to keep it? In my opinion, this is bug causing an undocumented side effect, because it renders the ``key`` parameter useless.
